src/components/cylc/tree/deltas.js

```diff
@@ -15,96 +15,62 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+import isArray from 'lodash/isArray'
 import {
+  createWorkflowNode,
   createCyclePointNode,
   createFamilyProxyNode,
   createJobNode,
   createTaskProxyNode
-} from '@/components/cylc/tree/tree-nodes'
-import { populateTreeFromGraphQLData } from '@/components/cylc/tree/index'
-import store from '@/store/index'
-import AlertModel from '@/model/Alert.model'
-
-/**
- * Helper object used to iterate pruned deltas data.
- */
-const PRUNED = {
-  jobs: 'removeJob',
-  taskProxies: 'removeTaskProxy',
-  familyProxies: 'removeFamilyProxy'
-}
-
-/**
- * @typedef {Object} DeltasPruned
- * @property {Array<string>} taskProxies - IDs of task proxies removed
- * @property {Array<string>} familyProxies - IDs of family proxies removed
- * @property {Array<string>} jobs - IDs of jobs removed
- */
-
-/**
- * Deltas pruned.
- *
- * @param {DeltasPruned} pruned - deltas pruned
- * @param {CylcTree} tree
- */
-function applyDeltasPruned (pruned, tree) {
-  Object.keys(PRUNED).forEach(prunedKey => {
-    if (pruned[prunedKey]) {
-      for (const id of pruned[prunedKey]) {
-        try {
-          tree[PRUNED[prunedKey]](id)
-        } catch (error) {
-          // eslint-disable-next-line no-console
-          console.error('Error applying pruned-delta, will continue processing the remaining data', error, id)
-          store.dispatch('setAlert', new AlertModel('Error applying pruned-delta, see browser console logs for more. Please reload your browser tab to retrieve the full flow state', null, 'error'))
-        }
-      }
-    }
-  })
-}
+} from '@/components/cylc/tree/nodes'
+import * as CylcTree from '@/components/cylc/tree/index'
```

Comment:
We cannot keep the state of the `CylcTree` in an object along with functions/methods. That's because we are now storing it in Vuex store, which doesn't allow functions/methods in the stored data :+1: The way the code is imported here, is very similar to how you can specify functions in a module in Python and call them directly. It looks like an Object, but is not really an instance with a constructor and local state.